log_outgoing_requests/validators.py

```diff
@@ -0,0 +1,54 @@
+from typing import TYPE_CHECKING, Any, Iterable, Optional
+
+if TYPE_CHECKING:  # pragma: nocover
+    from .models import OutgoingRequestsLogConfig
+
+
+def _validate_content_length(
+    content_length: Any, config: "OutgoingRequestsLogConfig"
+) -> bool:
+    max_content_length: Optional[int] = config.max_content_length
+    content_length_required: bool = config.content_length_required
+
+    # case 1: check of content_length not required
+    if not content_length_required:
+        return True
```

Comment:
W/r to content-length:

* You can have chunked transfers, when you basically receive an ongoing stream of (response) data. Then there will not be a `Content-Length` header, as it is unknown.
* Requests itself does not do any checks or logic based on the `Content-Length` response header (couldn't find references in the requests code), it just reads `CONTENT_CHUNK_SIZE` at a time and appends it. So even though a response may have a `Content-Length` set at whatever value, requests will probably just read everything anyway and `len(response.content)` will be the actual `bytes` size of the body. At this point, the damage is done and the entire response content has been loaded into memory.

So I think the best effort (but absolutely not fool-proof) we can do here is consider the header and if contains a value higher than our treshold -> do not log the body. If no header is set, proceed with logging, but perhaps we should then also emit `logger.warning` messages when `len(response.content) >= treshold` so we can at least track down the offenders and see what we can do about it.